### DISCUSSION TITLE
Use of end of life  AngularJS in Keycloak

### DISCUSSION THREAD

**Discussion by ss107-github** [keycloak/keycloak#21720] [Q&A] [answered]
Hi, 

I'm using Keycloak 12.x and it looks like Angular JS is used when the Administrator Console is called:
eg. <Keycloak URL>/auth/resources/gr1ve/common/keycloak/node_modules/angular-cookies/angular-cookies.min.js
It says Angular JS 1.8.0 in angular-cookies.min.js.

In keycloak-main\docs\dependency-license-information.md it also mentions AngularJS as one of the dependency, and is used in 
"themes/keycloak/common/resources/lib/angular".  

I checked the source code of Keycloak 22.0 and saw the same information in dependency-license-information.md.

Since Angular JS is officially end of support as of Jan 2022, therefore this is reported as a security vulnerability, I wonder if anyone can help me with the following questions:
1. Normally under what situation the AngularJS libraries are used in Keycloak? 
2. Do we have a plan to drop AngularJS and use the new Angular instead? 

Thanks in advance for your help

Regards,
SS

**Comment by jonkoops** [comment, marked as answer]
Angular JS support has been dropped already, user interfaces that were dependent on Angular JS have been replaced with new ones based on React.
